lib/src/date-input/DateInput.stories.tsx

```diff
@@ -107,45 +113,107 @@ export const Chromatic = () => (
   </>
 );
 
-const DatePicker = () => (
-  <ExampleContainer expanded>
-    <Title title="Show date picker" theme="light" level={4} />
-    <DxcDateInput label="Date input" defaultValue="06-04-2027" />
-  </ExampleContainer>
-);
+const DatePickerButtonStates = () => {
+  const colorsTheme: any = useTheme();
+  return (
+    <>
+      <ExampleContainer expanded>
+        <Title title="Show date picker over another element with z-index 0" theme="light" level={4} />
```

Comment:
I think you didn't understand me Jiale... What I meant is that we need to test that the date picker appears over other web elements with a `z-index` > 0. It's good to check that it appears over our components but the real case is with items like Dialogs, Headers, etc, whose `z-index` is higher than expected. For example, in the Autosuggest:

```
      <ExampleContainer>
        <Title title="Autosuggest listbox" theme="light" level={2} />
        <ExampleContainer>
          <Title title="List dialog uses a Radix Popover to appear over elements with a certain z-index" theme="light" level={3} />
          <div
            style={{
              display: "flex",
              flexDirection: "column",
              gap: "20px",
              height: "150px",
              width: "500px",
              marginBottom: "250px",
              padding: "20px",
              border: "1px solid black",
              borderRadius: "4px",
              overflow: "auto",
              zIndex: "1300",
            }}
          >
            <DxcTextInput
              label="Label"
              suggestions={countries}
              optional
              placeholder="Choose an option"
              size="fillParent"
            />
            <button style={{ zIndex: "1", width: "100px" }}>Submit</button>
          </div>
        </ExampleContainer>
```